test/sql/chunk_publication.sql

```diff
@@ -0,0 +1,322 @@
+-- This file and its contents are licensed under the Apache License 2.0.
+-- Please see the included NOTICE for copyright information and
+-- LICENSE-APACHE for a copy of the license.
+
+-- Test automatic addition of chunks to publications
+-- Publications require superuser privileges
+
+\c :TEST_DBNAME :ROLE_SUPERUSER
+
+SET client_min_messages = WARNING;
+
+-- Test 1: Basic single publication
+CREATE TABLE test_hypertable (time timestamptz NOT NULL, device_id int, value float, extra text);
+SELECT create_hypertable('test_hypertable', 'time', chunk_time_interval => interval '1 day');
+
+-- Insert to create first chunk
+INSERT INTO test_hypertable VALUES ('2024-01-01 00:00:00+00', 1, 1.0, 'data1');
+
+-- Create publication and add hypertable
+CREATE PUBLICATION test_pub FOR TABLE test_hypertable;
+
+-- Verify initial state (1 chunk)
+SELECT schemaname, tablename, attnames, rowfilter FROM pg_publication_tables WHERE pubname = 'test_pub' ORDER BY schemaname, tablename;
+
+-- Insert to create 2 more chunks (total 3 chunks)
+INSERT INTO test_hypertable VALUES ('2024-01-02 00:00:00+00', 2, 2.0, 'data2');
+INSERT INTO test_hypertable VALUES ('2024-01-03 00:00:00+00', 3, 3.0, 'data3');
+
+-- Verify state (3 chunks)
+SELECT schemaname, tablename, attnames, rowfilter FROM pg_publication_tables WHERE pubname = 'test_pub' ORDER BY schemaname, tablename;
+
+-- Insert to create 5 more chunks (total 8 chunks)
+INSERT INTO test_hypertable VALUES ('2024-01-04 00:00:00+00', 4, 4.0, 'data4');
+INSERT INTO test_hypertable VALUES ('2024-01-05 00:00:00+00', 5, 5.0, 'data5');
+INSERT INTO test_hypertable VALUES ('2024-01-06 00:00:00+00', 6, 6.0, 'data6');
+INSERT INTO test_hypertable VALUES ('2024-01-07 00:00:00+00', 7, 7.0, 'data7');
+INSERT INTO test_hypertable VALUES ('2024-01-08 00:00:00+00', 8, 8.0, 'data8');
+
+-- Verify final state (8 chunks)
+SELECT schemaname, tablename, attnames, rowfilter FROM pg_publication_tables WHERE pubname = 'test_pub' ORDER BY schemaname, tablename;
+
+-- Verify chunk removal via DROP TABLE
```

Comment:
@gayyappan Here is a test that covers drop chunk in 3 scenarios.
1) drop_chunk SQL api
2) DROP TABLE chunk_table
3) TRUNCATE TABLE